src/Ilios/CoreBundle/Entity/Manager/PermissionManager.php

```diff
@@ -125,6 +125,58 @@ public function userHasReadPermissionToSchool(UserInterface $user, SchoolInterfa
     /**
      * {@inheritdoc}
      */
+    public function userHasReadPermissionToSchools(UserInterface $user, ArrayCollection $schools)
+    {
+        $criteria = [
+            'tableName'     => 'schools',
+            self::CAN_READ  => true,
+            'user'          => $user,
+        ];
+
+        $schoolsWithReadPermission = $this->findPermissionBy($criteria);
+        
+        if ($schoolsWithReadPermission->count() === 0) {
+            return false;
+        }
+        
+        foreach ($schools as $school) {
```

Comment:
the loop could be replaced with:

```php
$overlap = array_intersect($schools->toArray(), $schoolsWithReadPermissions->toArray());
return ! empty($overlap);
```

shouldn't really matter tho, since we're dealing with only a few records... just a thought.